test/powershell/Modules/Microsoft.PowerShell.Utility/Get-SecureRandom.Tests.ps1

```diff
@@ -0,0 +1,183 @@
+        $randomNumber = Get-SecureRandom -InputObject 1, 2, 3, 5, 8, 13
+        $randomNumber | Should -BeIn 1, 2, 3, 5, 8, 13
+    }
+
+    It "Should return an array " {
+        $randomNumber = Get-SecureRandom -InputObject 1, 2, 3, 5, 8, 13 -Count 3
+        $randomNumber.Count | Should -Be 3
+        ,$randomNumber | Should -BeOfType System.Array
+    }
+
+    It "Should return three random numbers for array of 1,2,3,5,8,13 " {
+        $randomNumber = Get-SecureRandom -InputObject 1, 2, 3, 5, 8, 13 -Count 3
+        $randomNumber.Count | Should -Be 3
+        $randomNumber[0] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[1] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[2] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[3] | Should -BeNullOrEmpty
+    }
+
+    It "Should return all the numbers for array of 1,2,3,5,8,13 in no particular order" {
+        $randomNumber = Get-SecureRandom -InputObject 1, 2, 3, 5, 8, 13 -Count ([int]::MaxValue)
+        $randomNumber.Count | Should -Be 6
+        $randomNumber[0] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[1] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[2] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[3] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[4] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[5] | Should -BeIn 1, 2, 3, 5, 8, 13
+        $randomNumber[6] | Should -BeNullOrEmpty
+    }
+
+    It "Should return all the numbers for array of 1,2,3,5,8,13 in randomized order when the Shuffle switch is used" {
+        $randomNumber = Get-SecureRandom -InputObject 1, 2, 3, 5, 8, 13 -Shuffle
+        $randomNumber.Count | Should -Be 6
+        $randomNumber | Should -BeIn 1, 2, 3, 5, 8, 13
+    }
+
+    It "Should return for a string collection " {
+        $randomNumber = Get-SecureRandom -InputObject "red", "yellow", "blue"
+        $randomNumber | Should -Be ("red" -or "yellow" -or "blue")
+    }
+
+    It "Should return a number for hexadecimal " {
+        $randomNumber = Get-SecureRandom 0x07FFFFFFFFF
+        $randomNumber | Should -BeLessThan 549755813887
+        $randomNumber | Should -BeGreaterThan 0
+    }
+
+    It "Should throw an error because the hexadecimal number is to large " {
+        { Get-SecureRandom 0x07FFFFFFFFFFFFFFFF } | Should -Throw "Value was either too large or too small for a UInt32"
+    }
+
+    It "Should accept collection containing empty string for -InputObject" {
+        1..10 | ForEach-Object {
+            Get-SecureRandom -InputObject @('a','b','') | Should -BeIn 'a','b',''
+        }
+    }
+
+    It "Should accept `$null in collection for -InputObject" {
+        1..10 | ForEach-Object {
+            Get-SecureRandom -InputObject @('a','b',$null) | Should -BeIn 'a','b',$null
+        }
+    }
+
+    It 'Should not have a -SetSeed parameter' {
+        (get-command -ParameterName SetSeed).Name | Should -Not -Contain 'Get-SecureRandom'
```

Comment:
I should think this would be more direct as:
```suggestion
        (Get-command Get-SecureRandom).Parameters['SetSeed'] | Should -BeNullOrEmpty
```